### ISSUE TITLE
unwrap_used not triggered when chained from get() or get_mut()

### ISSUE THREAD

**Issue by doy** [rust-lang/rust-clippy#8124]
### Summary

it appears that the `unwrap_used` lint does not trigger for code of the form `foo.get(i).unwrap()` or `foo.get_mut(i).unwrap()`

### Lint Name

unwrap_used

### Reproducer

I tried this code:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2ed9ce35f88fd2d4c16151aeed566134

I expected to see this happen:

all three functions should trigger the `unwrap_used` lint

Instead, this happened:
the second function does not trigger the lint

### Version

```text
rustc 1.57.0 (f1edd0429 2021-11-29)
binary: rustc
commit-hash: f1edd0429582dd29cccacaf50fd134b05593bd9c
commit-date: 2021-11-29
host: x86_64-unknown-linux-gnu
release: 1.57.0
LLVM version: 13.0.0
```

**Comment by giraffate** [comment]
We have the [`get_unwrap`](https://rust-lang.github.io/rust-clippy/master/index.html#get_unwrap) lint, so can you use this?

**Comment by doy** [comment]
i could, but this still feels like very surprising behavior

**Comment by giraffate** [comment]
It might be better to merge `unwrap_used` and `get_unwrap` like [this](https://github.com/rust-lang/rust-clippy/pull/5563). It looks good to post on [the Clippy zulip channel](https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy) to get other's opinions.

**Comment by giraffate** [comment]
It was discussed here: https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Merge.20.60get_unwrap.60.20and.20.60unwrap_used.60
